### PR TITLE
Migrate `DeleteDelegateDto` validation to `zod`

### PR DESCRIPTION
**Pull request by hectorgomezv**
## Summary

This migrates the validation of `DeleteDelegateDto` to `zod`.

## Changes

- Remove `DeleteDelegateDtoValidationPipe` and associated schema.
- Add `DeleteDelegateDtoSchema`.
- Verifiy the associated endpoint data validation by `ValidationPipe(DeleteDelegateDtoSchema)`.
- Update associated tests.

### CODE REVIEW

**Review by iamacook — APPROVED** [review]

**Review by iamacook — APPROVED** [review]

**Comment by iamacook** [comment]
I'm marking this as a draft as we need add tests for it.

**Comment by hectorgomezv** [comment]
I've added the validation schema unit tests in [98d6b26](https://github.com/safe-global/safe-client-gateway/pull/1265/commits/98d6b26408ab8d0097263923c979b3d7b9db2695)

**Review by iamacook — APPROVED** [review]

File: src/datasources/transaction-api/transaction-api.service.ts [hunk 1]
```diff
@@ -200,11 +200,11 @@ export class TransactionApi implements ITransactionApi {
...
-    safeAddress?: string;
-    delegate?: string;
-    delegator?: string;
-    signature?: string;
-    label?: string;
+    safeAddress: string | null;
+    delegate: string;
+    delegator: string;
+    signature: string;
+    label: string;
```
Comment (iamacook) [inline src/datasources/transaction-api/transaction-api.service.ts:207]: Is this branched off #1275?

Comment (iamacook) [reply src/datasources/transaction-api/transaction-api.service.ts:207]: The only comments I have regarding this PR, I left there.

**Pull request merged**